atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/TransactionCacheValueStoreImpl.java

```diff
@@ -98,15 +99,17 @@ public TransactionCacheValueStore createWithFilteredSnapshot(CommitUpdate commit
 
     @Override
     public Map<Cell, CacheValue> getCachedValues(TableReference table, Set<Cell> cells) {
-        Map<Cell, CacheValue> locallyCachedValues = getLocallyCachedValues(table, cells);
+        Set<Cell> locallyWrittenCells = getLocallyWrittenCells(table, cells);
+        Set<Cell> cacheableCells = Sets.difference(cells, locallyWrittenCells);
+
+        Map<Cell, CacheValue> locallyCachedValues = getLocallyCachedValues(table, cacheableCells);
 
         // Filter out which values have not been read yet
-        Set<Cell> remainingCells = Sets.difference(cells, locallyCachedValues.keySet());
+        Set<Cell> remainingCells = Sets.difference(cacheableCells, locallyCachedValues.keySet());
 
         // Read values from the snapshot. For the hits, mark as hit in the local map.
         Map<Cell, CacheValue> snapshotCachedValues = getSnapshotValues(table, remainingCells);
-        snapshotCachedValues.forEach(
-                (cell, value) -> localUpdates.put(CellReference.of(table, cell), LocalCacheEntry.hit(value)));
+        snapshotCachedValues.forEach((cell, value) -> cacheHitInternal(table, cell, value));
```

Comment:
This is the only functional change in this method, right? Reading writes and reads from the cache separately does not seem to do anything special

Reply:
The only change is just that we're filtering out the writes entirely (because filtering them out from `localUpdates` isn't enough). The `cacheHitInternal` is just an extra